src/Microsoft.SqlTools.ServiceLayer/DacFx/DacFxDeployOperation.cs

```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.SqlServer.Dac;
+using Microsoft.SqlTools.ServiceLayer.DacFx.Contracts;
+using Microsoft.SqlTools.ServiceLayer.TaskServices;
+using Microsoft.SqlTools.Utility;
+using System;
+using System.Data.SqlClient;
+using System.Diagnostics;
+
+namespace Microsoft.SqlTools.ServiceLayer.DacFx
+{
+    /// <summary>
+    /// Class to represent an in-progress deploy operation
+    /// </summary>
+    class DacFxDeployOperation : DacFxOperation
+    {
+        public DacFxDeployParams Parameters { get; }
+        private SqlConnection sqlConnection { get; set; }
+
+        public DacFxDeployOperation(DacFxDeployParams parameters, SqlConnection sqlConnection)
+        {
+            Validate.IsNotNull("parameters", parameters);
+            Validate.IsNotNull("sqlConnection", sqlConnection);
+            this.Parameters = parameters;
+            this.sqlConnection = sqlConnection;
+        }
+
+        public override void Execute(TaskExecutionMode mode)
+        {
+            if (this.CancellationToken.IsCancellationRequested)
+            {
+                throw new OperationCanceledException(this.CancellationToken);
+            }
+
+            try
+            {
+                DacServices ds = new DacServices(this.sqlConnection.ConnectionString);
```

Comment:
What about progress and message events? We want clients to see these as dacfx operations will be long-running.

Reply:
I added a handler for ProgressChanged, but it doesn't do anything since there isn't a way for the progress to be shown in ADS yet.